### PR TITLE
Removes check for MapReduce usage on NATIVE storage

### PR DESCRIPTION
**Pull request by ahmetmircik**
Check for NATIVE storage is moved to ee.
fixes https://github.com/hazelcast/hazelcast-enterprise/issues/926

tests are here: https://github.com/hazelcast/hazelcast-enterprise/pull/966

### CODE REVIEW

File: hazelcast-client/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java [hunk 1]
```diff
@@ -121,7 +121,7 @@ private ClientProxyFactory createClientMapProxyFactory() {
             public ClientProxy create(String id) {
                 NearCacheConfig nearCacheConfig = client.getClientConfig().getNearCacheConfig(id);
                 if (nearCacheConfig != null) {
-                    checkInMemoryFormat(nearCacheConfig.getInMemoryFormat());
+                    checkNotNATIVE(nearCacheConfig.getInMemoryFormat());
                     return new NearCachedClientMapProxy(MapService.SERVICE_NAME, id);
```
Comment (tombujok) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java]: Looks good. It just bothers me that Native is all uppercase, thus does not comply with Java convention ;)

Comment (ahmetmircik) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java]: fixed

**Comment by vbekiaris** [comment]
👍

**Comment by tombujok** [comment]
:+1:

**Pull request merged**